File: CodeSamples/python/swagger_client/models/check_transaction_settings.py
# coding: utf-8

class CheckTransactionSettings(object):
    swagger_types = {
        'reader': 'FieldInfo',
        'contact_name': 'FieldInfo',
        'contact_title': 'FieldInfo',
        'email_address': 'FieldInfo',
        'phone_default': 'str',
        'fax': 'FieldInfo',
        'billing_address_default': 'str',
        'billing_address2': 'FieldInfo',
        'billing_city_default': 'str',
        'billing_state_default': 'str',
        'billing_zip_default': 'str',
        'po_number': 'FieldInfo',
        'invoice_number': 'FieldInfo',
        'business_name': 'FieldInfo',
        'customer_number': 'FieldInfo',
        'reference_note': 'FieldInfo',
        'default_sec_code': 'str'
    }

    attribute_map = {
        'reader': 'Reader',
        'contact_name': 'ContactName',
        'contact_title': 'ContactTitle',
        'email_address': 'EmailAddress',
        'phone_default': 'PhoneDefault',
        'fax': 'Fax',
        'billing_address_default': 'BillingAddressDefault',
        'billing_address2': 'BillingAddress2',
        'billing_city_default': 'BillingCityDefault',
        'billing_state_default': 'BillingStateDefault',
        'billing_zip_default': 'BillingZipDefault',
        'po_number': 'PONumber',
        'invoice_number': 'InvoiceNumber',
        'business_name': 'BusinessName',
        'customer_number': 'CustomerNumber',
        'reference_note': 'ReferenceNote',
        'default_sec_code': 'DefaultSECCode'
    }

    def __init__(self, reader=None, contact_name=None, contact_title=None, email_address=None, phone_default=None, fax=None, billing_address_default=None, billing_address2=None, billing_city_default=None, billing_state_default=None, billing_zip_default=None, po_number=None, invoice_number=None, business_name=None, customer_number=None, reference_note=None, default_sec_code=None):  # noqa: E501
        """CheckTransactionSettings - a model defined in Swagger"""  # noqa: E501
        self._reader = None
        self._contact_name = None
        self._contact_title = None
        self._email_address = None
        self._phone_default = None
        self._fax = None
        self._billing_address_default = None
        self._billing_address2 = None
        self._billing_city_default = None
        self._billing_state_default = None
        self._billing_zip_default = None
        self._po_number = None
        self._invoice_number = None
        self._business_name = None
        self._customer_number = None
        self._reference_note = None
        self._default_sec_code = None
        self.discriminator = None
        if reader is not None:
            self.reader = reader
        if contact_name is not None:
            self.contact_name = contact_name
        if contact_title is not None:
            self.contact_title = contact_title
        if email_address is not None:
            self.email_address = email_address
        if phone_default is not None:
            self.phone_default = phone_default
        if fax is not None:
            self.fax = fax
        if billing_address_default is not None:
            self.billing_address_default = billing_address_default
        if billing_address2 is not None:
            self.billing_address2 = billing_address2
        if billing_city_default is not None:
            self.billing_city_default = billing_city_default
        if billing_state_default is not None:
            self.billing_state_default = billing_state_default
        if billing_zip_default is not None:
            self.billing_zip_default = billing_zip_default
        if po_number is not None:
            self.po_number = po_number
        if invoice_number is not None:
            self.invoice_number = invoice_number
        if business_name is not None:
            self.business_name = business_name
        if customer_number is not None:
            self.customer_number = customer_number
        if reference_note is not None:
            self.reference_note = reference_note
        if default_sec_code is not None:
            self.default_sec_code = default_sec_code

    @property
    def reader(self):
        return self._reader

    @reader.setter
    def reader(self, reader):
        self._reader = reader

    @property
    def contact_name(self):
        return self._contact_name

    @contact_name.setter
    def contact_name(self, contact_name):
        self._contact_name = contact_name

    @property
    def contact_title(self):
        return self._contact_title

    @contact_title.setter
    def contact_title(self, contact_title):
        self._contact_title = contact_title

    @property
    def email_address(self):
        return self._email_address

    @email_address.setter
    def email_address(self, email_address):
        self._email_address = email_address

    @property
    def phone_default(self):
        return self._phone_default

    @phone_default.setter
    def phone_default(self, phone_default):
        self._phone_default = phone_default

    @property
    def fax(self):
        return self._fax

    @fax.setter
    def fax(self, fax):
        self._fax = fax

    @property
    def billing_address_default(self):
        return self._billing_address_default

    @billing_address_default.setter
    def billing_address_default(self, billing_address_default):
        self._billing_address_default = billing_address_default

    @property
    def billing_address2(self):
        return self._billing_address2

    @billing_address2.setter
    def billing_address2(self, billing_address2):
        self._billing_address2 = billing_address2

    @property
    def billing_city_default(self):
        return self._billing_city_default

    @billing_city_default.setter
    def billing_city_default(self, billing_city_default):
        self._billing_city_default = billing_city_default

    @property
    def billing_state_default(self):
        return self._billing_state_default

    @billing_state_default.setter
    def billing_state_default(self, billing_state_default):
        self._billing_state_default = billing_state_default

    @property
    def billing_zip_default(self):
        return self._billing_zip_default

    @billing_zip_default.setter
    def billing_zip_default(self, billing_zip_default):
        self._billing_zip_default = billing_zip_default

    @property
    def po_number(self):
        return self._po_number

    @po_number.setter
    def po_number(self, po_number):
        self._po_number = po_number

    @property
    def invoice_number(self):
        return self._invoice_number

    @invoice_number.setter
    def invoice_number(self, invoice_number):
        self._invoice_number = invoice_number

    @property
    def business_name(self):
        return self._business_name

    @business_name.setter
    def business_name(self, business_name):
        self._business_name = business_name

    @property
    def customer_number(self):
        return self._customer_number

    @customer_number.setter
    def customer_number(self, customer_number):
        self._customer_number = customer_number

    @property
    def reference_note(self):
        return self._reference_note

    @reference_note.setter
    def reference_note(self, reference_note):
        self._reference_note = reference_note

    @property
    def default_sec_code(self):
        return self._default_sec_code

    @default_sec_code.setter
    def default_sec_code(self, default_sec_code):
        self._default_sec_code = default_sec_code

File: CodeSamples/php/SwaggerClient-php/lib/Model/CheckTransactionSettings.php
<?php

namespace Swagger\Client\Model;

use \ArrayAccess;
use \Swagger\Client\ObjectSerializer;

class CheckTransactionSettings implements ModelInterface, ArrayAccess
{
    const DISCRIMINATOR = null;

    protected static $swaggerModelName = 'CheckTransactionSettings';

    protected static $swaggerTypes = [
        'reader' => '\Swagger\Client\Model\FieldInfo',
        'contact_name' => '\Swagger\Client\Model\FieldInfo',
        'contact_title' => '\Swagger\Client\Model\FieldInfo',
        'email_address' => '\Swagger\Client\Model\FieldInfo',
        'phone_default' => 'string',
        'fax' => '\Swagger\Client\Model\FieldInfo',
        'billing_address_default' => 'string',
        'billing_address2' => '\Swagger\Client\Model\FieldInfo',
        'billing_city_default' => 'string',
        'billing_state_default' => 'string',
        'billing_zip_default' => 'string',
        'po_number' => '\Swagger\Client\Model\FieldInfo',
        'invoice_number' => '\Swagger\Client\Model\FieldInfo',
        'business_name' => '\Swagger\Client\Model\FieldInfo',
        'customer_number' => '\Swagger\Client\Model\FieldInfo',
        'reference_note' => '\Swagger\Client\Model\FieldInfo',
        'default_sec_code' => 'string'
    ];

    protected static $swaggerFormats = [
        'reader' => null,
        'contact_name' => null,
        'contact_title' => null,
        'email_address' => null,
        'phone_default' => null,
        'fax' => null,
        'billing_address_default' => null,
        'billing_address2' => null,
        'billing_city_default' => null,
        'billing_state_default' => null,
        'billing_zip_default' => null,
        'po_number' => null,
        'invoice_number' => null,
        'business_name' => null,
        'customer_number' => null,
        'reference_note' => null,
        'default_sec_code' => null
    ];

    public static function swaggerTypes()
    {
        return self::$swaggerTypes;
    }

    public static function swaggerFormats()
    {
        return self::$swaggerFormats;
    }

    protected static $attributeMap = [
        'reader' => 'Reader',
        'contact_name' => 'ContactName',
        'contact_title' => 'ContactTitle',
        'email_address' => 'EmailAddress',
        'phone_default' => 'PhoneDefault',
        'fax' => 'Fax',
        'billing_address_default' => 'BillingAddressDefault',
        'billing_address2' => 'BillingAddress2',
        'billing_city_default' => 'BillingCityDefault',
        'billing_state_default' => 'BillingStateDefault',
        'billing_zip_default' => 'BillingZipDefault',
        'po_number' => 'PONumber',
        'invoice_number' => 'InvoiceNumber',
        'business_name' => 'BusinessName',
        'customer_number' => 'CustomerNumber',
        'reference_note' => 'ReferenceNote',
        'default_sec_code' => 'DefaultSECCode'
    ];

    protected static $setters = [
        'reader' => 'setReader',
        'contact_name' => 'setContactName',
        'contact_title' => 'setContactTitle',
        'email_address' => 'setEmailAddress',
        'phone_default' => 'setPhoneDefault',
        'fax' => 'setFax',
        'billing_address_default' => 'setBillingAddressDefault',
        'billing_address2' => 'setBillingAddress2',
        'billing_city_default' => 'setBillingCityDefault',
        'billing_state_default' => 'setBillingStateDefault',
        'billing_zip_default' => 'setBillingZipDefault',
        'po_number' => 'setPoNumber',
        'invoice_number' => 'setInvoiceNumber',
        'business_name' => 'setBusinessName',
        'customer_number' => 'setCustomerNumber',
        'reference_note' => 'setReferenceNote',
        'default_sec_code' => 'setDefaultSecCode'
    ];

    protected static $getters = [
        'reader' => 'getReader',
        'contact_name' => 'getContactName',
        'contact_title' => 'getContactTitle',
        'email_address' => 'getEmailAddress',
        'phone_default' => 'getPhoneDefault',
        'fax' => 'getFax',
        'billing_address_default' => 'getBillingAddressDefault',
        'billing_address2' => 'getBillingAddress2',
        'billing_city_default' => 'getBillingCityDefault',
        'billing_state_default' => 'getBillingStateDefault',
        'billing_zip_default' => 'getBillingZipDefault',
        'po_number' => 'getPoNumber',
        'invoice_number' => 'getInvoiceNumber',
        'business_name' => 'getBusinessName',
        'customer_number' => 'getCustomerNumber',
        'reference_note' => 'getReferenceNote',
        'default_sec_code' => 'getDefaultSecCode'
    ];

    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    public static function setters()
    {
        return self::$setters;
    }

    public static function getters()
    {
        return self::$getters;
    }

    public function getModelName()
    {
        return self::$swaggerModelName;
    }

    protected $container = [];

    public function __construct(array $data = null)
    {
        $this->container['reader'] = isset($data['reader']) ? $data['reader'] : null;
        $this->container['contact_name'] = isset($data['contact_name']) ? $data['contact_name'] : null;
        $this->container['contact_title'] = isset($data['contact_title']) ? $data['contact_title'] : null;
        $this->container['email_address'] = isset($data['email_address']) ? $data['email_address'] : null;
        $this->container['phone_default'] = isset($data['phone_default']) ? $data['phone_default'] : null;
        $this->container['fax'] = isset($data['fax']) ? $data['fax'] : null;
        $this->container['billing_address_default'] = isset($data['billing_address_default']) ? $data['billing_address_default'] : null;
        $this->container['billing_address2'] = isset($data['billing_address2']) ? $data['billing_address2'] : null;
        $this->container['billing_city_default'] = isset($data['billing_city_default']) ? $data['billing_city_default'] : null;
        $this->container['billing_state_default'] = isset($data['billing_state_default']) ? $data['billing_state_default'] : null;
        $this->container['billing_zip_default'] = isset($data['billing_zip_default']) ? $data['billing_zip_default'] : null;
        $this->container['po_number'] = isset($data['po_number']) ? $data['po_number'] : null;
        $this->container['invoice_number'] = isset($data['invoice_number']) ? $data['invoice_number'] : null;
        $this->container['business_name'] = isset($data['business_name']) ? $data['business_name'] : null;
        $this->container['customer_number'] = isset($data['customer_number']) ? $data['customer_number'] : null;
        $this->container['reference_note'] = isset($data['reference_note']) ? $data['reference_note'] : null;
        $this->container['default_sec_code'] = isset($data['default_sec_code']) ? $data['default_sec_code'] : null;
    }

    public function listInvalidProperties()
    {
        $invalidProperties = [];

        return $invalidProperties;
    }

    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    public function getReader()
    {
        return $this->container['reader'];
    }

    public function setReader($reader)
    {
        $this->container['reader'] = $reader;

        return $this;
    }

    public function getContactName()
    {
        return $this->container['contact_name'];
    }

    public function setContactName($contact_name)
    {
        $this->container['contact_name'] = $contact_name;

        return $this;
    }

    public function getContactTitle()
    {
        return $this->container['contact_title'];
    }

    public function setContactTitle($contact_title)
    {
        $this->container['contact_title'] = $contact_title;

        return $this;
    }

    public function getEmailAddress()
    {
        return $this->container['email_address'];
    }

    public function setEmailAddress($email_address)
    {
        $this->container['email_address'] = $email_address;

        return $this;
    }

    public function getPhoneDefault()
    {
        return $this->container['phone_default'];
    }

    public function setPhoneDefault($phone_default)
    {
        $this->container['phone_default'] = $phone_default;

        return $this;
    }

    public function getFax()
    {
        return $this->container['fax'];
    }

    public function setFax($fax)
    {
        $this->container['fax'] = $fax;

        return $this;
    }

    public function getBillingAddressDefault()
    {
        return $this->container['billing_address_default'];
    }

    public function setBillingAddressDefault($billing_address_default)
    {
        $this->container['billing_address_default'] = $billing_address_default;

        return $this;
    }

    public function getBillingAddress2()
    {
        return $this->container['billing_address2'];
    }

    public function setBillingAddress2($billing_address2)
    {
        $this->container['billing_address2'] = $billing_address2;

        return $this;
    }

    public function getBillingCityDefault()
    {
        return $this->container['billing_city_default'];
    }

    public function setBillingCityDefault($billing_city_default)
    {
        $this->container['billing_city_default'] = $billing_city_default;

        return $this;
    }

    public function getBillingStateDefault()
    {
        return $this->container['billing_state_default'];
    }

    public function setBillingStateDefault($billing_state_default)
    {
        $this->container['billing_state_default'] = $billing_state_default;

        return $this;
    }

    public function getBillingZipDefault()
    {
        return $this->container['billing_zip_default'];
    }

    public function setBillingZipDefault($billing_zip_default)
    {
        $this->container['billing_zip_default'] = $billing_zip_default;

        return $this;
    }

    public function getPoNumber()
    {
        return $this->container['po_number'];
    }

    public function setPoNumber($po_number)
    {
        $this->container['po_number'] = $po_number;

        return $this;
    }

    public function getInvoiceNumber()
    {
        return $this->container['invoice_number'];
    }

    public function setInvoiceNumber($invoice_number)
    {
        $this->container['invoice_number'] = $invoice_number;

        return $this;
    }

    public function getBusinessName()
    {
        return $this->container['business_name'];
    }

    public function setBusinessName($business_name)
    {
        $this->container['business_name'] = $business_name;

        return $this;
    }

    public function getCustomerNumber()
    {
        return $this->container['customer_number'];
    }

    public function setCustomerNumber($customer_number)
    {
        $this->container['customer_number'] = $customer_number;

        return $this;
    }

    public function getReferenceNote()
    {
        return $this->container['reference_note'];
    }

    public function setReferenceNote($reference_note)
    {
        $this->container['reference_note'] = $reference_note;

        return $this;
    }

    public function getDefaultSecCode()
    {
        return $this->container['default_sec_code'];
    }

    public function setDefaultSecCode($default_sec_code)
    {
        $this->container['default_sec_code'] = $default_sec_code;

        return $this;
    }
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    public function offsetGet($offset)
    {
        return isset($this->container[$offset]) ? $this->container[$offset] : null;
    }

    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }

    public function __toString()
    {
        if (defined('JSON_PRETTY_PRINT')) { // use JSON pretty print
            return json_encode(
                ObjectSerializer::sanitizeForSerialization($this),
                JSON_PRETTY_PRINT
            );
        }

        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}
